testing/acceptance_test.py

```diff
@@ -1584,3 +1584,29 @@ def test_no_terminal_plugin(pytester: Pytester) -> None:
     pytester.makepyfile("def test(): assert 1 == 2")
     result = pytester.runpytest("-pno:terminal", "-s")
     assert result.ret == ExitCode.TESTS_FAILED
+
+
+def test_get_exception_on_teardown_failure(pytester: Pytester) -> None:
+    """Smoke test to be sure teardown exceptions handled properly via node property"""
+    pytester.makepyfile(
+        conftest="""
+        import sys
+        import pytest
+        def pytest_exception_interact(node, call, report):
+            sys.stderr.write("{}".format(node.teardown_exceptions))
+
+        import pytest
+        @pytest.fixture
+        def mylist():
+            yield
+            raise AssertionError(111)
+        """,
+        test_file="""
+        def test_func(mylist):
+            assert True
+        """,
+    )
+    result = pytester.runpytest()
+    assert result.ret == ExitCode.TESTS_FAILED
+    assert "AssertionError(111)" in result.stderr.str()
+    result.stdout.fnmatch_lines(["*1 error*"])
```

Comment:
this looks a bit unreliable to me, `assert "AssertionError(111)" in result.stderr.str()` could plausibly pass for other reasons than `pytest_exception_interact` printing it. Printing some marker characters surrounding it should suffice though, e.g. ```assert "node.teardown_exceptions: `AssertionError(111)`" in result.stderr.str()```.

It's also better to use `result.assert_outcomes(errors=1)` than fnmatch'ing lines

Reply:
I've put this stuff because there is unresolved one https://github.com/pytest-dev/pytest/issues/9909 
btw @jakkdl please, share you mind on that: https://github.com/pytest-dev/pytest/pull/13002/commits/a8e0aba5292e89c30d3b1fa266af69cf193a5723#diff-c2ee4e81db455299781df03980cd131ee4680fa5b6e5e484f3355dbeef8452c6R1611

Reply:
Nothing to comment, the PR doesn't touch the pytest architecture.